Source/Core/DolphinQt/MainWindow.cpp

```diff
@@ -1565,30 +1565,28 @@ void MainWindow::OnStopRecording()
 {
   if (Movie::IsRecordingInput())
     OnExportRecording();
-
-  Movie::EndPlayInput(false);
-  emit RecordingStatusChanged(true);
+  if (Movie::IsMovieActive())
+    Movie::EndPlayInput(false);
+  emit RecordingStatusChanged(false);
 }
 
 void MainWindow::OnExportRecording()
 {
   bool was_paused = Core::GetState() == Core::State::Paused;
 
-  if (was_paused)
+  if (!was_paused)
     Core::SetState(Core::State::Paused);
 
   QString dtm_file = QFileDialog::getSaveFileName(this, tr("Select the Recording File"), QString(),
                                                   tr("Dolphin TAS Movies (*.dtm)"));
 
-  if (was_paused)
-    Core::SetState(Core::State::Running);
-
   if (dtm_file.isEmpty())
     return;
 
-  Core::SetState(Core::State::Running);
-
   Movie::SaveRecording(dtm_file.toStdString());
+
+  if (!was_paused)
+    Core::SetState(Core::State::Running);
```

Comment:
This will not be called if `dtm_file.isEmpty()` hits. I don't think this is intended?

Reply:
The main reason I did it was in order to match the DolphinWX logic: https://github.com/dolphin-emu/dolphin/blob/5.0/Source/Core/DolphinWX/FrameTools.cpp#L1262-L1268

It makes some sense to me – that way you get a second chance to save at the exact same point in case you pressed Cancel or something. But I don't think the other way of doing it is illogical either. (If someone was in the running state before pressing Export Recording, I don't think they could've pressed Export Recording with much precision, so maybe they don't care about the exact point in time they save the DTM...)

Reply:
As long as it doesn't unpause if you had Dolphin paused initially, I'm fine with having different behavior than what's currently in the PR. Would you like me to change it?

Reply:
Hm, I see your point. But at the same time, we _automatically_ pause but never unpause that way.
With that I don't really lean towards any particular resolution; but I'd like to see a comment if we decide to leave it this way (just to point out "hey, this is intentional" inb4 future issue reports)

Reply:
Changed it for now. There was no reason specified in the DolphinWX source code, so I don't feel that I have any good comment to write...